Feature: separate CSS and JS from cheerio body refactoring

PiiXiieeS opened the pull request:
- Updated Highlightjs to version 11.4.0 and split injection of styles and javascript
- step addHighlight changed to fixCode
- 2 new steps addLibrariesCSS and addLibrariesJS
- move JS and activation of zooming to addLibrariesJS
- add dark and light theme for highlightjs
- improve testing fixDrawio
- move iframe-resizer to addLibrariesJS
- getFirstExcerpt to load excerpt in JSON field via API
- parseHeaderBlog to load excerpt and image in JSON field via API
- split css and js from slides process
- CSS moved to addSlidesCSS and loaded from cdnjs
- JS moved to addSlidesJS and loaded from cdnjs
- addSlides remains now headless

resolves #284, #295

PiiXiieeS committed:
feat(highlightjs): separate CSS and JS from code

PiiXiieeS commented:
@MeStrak started with the Highlightjs denormalization of libraries ;-)

MeStrak commented:
> @MeStrak started with the Highlightjs denormalization of libraries ;-)

Great! I reverted the conditional css/js injection in the API branch.

PiiXiieeS committed:
test: fixCode unit test

PiiXiieeS committed:
feat: move zooming to addLibraries

- move js and activation of zooming to addLibrariesJS
- add dark and light theme for highlightjs
- improve testing fixDrawio

PiiXiieeS committed:
feat: move iframe-resizer to addLibrariesJS

- getFirstExcerpt to load excerpt in JSON field via API
- parseHeaderBlog to load excerpt and image in JSON field via API

PiiXiieeS committed:
fix(fixCode): avoid unscaped html in code blocks

PiiXiieeS committed:
build: remove local libraries

PiiXiieeS committed:
docs: upgrade docs to vuepress 2.0

PiiXiieeS committed:
fix(media): avoid very large images

Avoid very large images when  they shall be small because
they have been resized.

PiiXiieeS committed:
feat(slides): split css and js from slides process

- CSS moved to addSlidesCSS and loaded from cdnjs
- JS moved to addSlidesJS and loaded from cdnjs
- addSlides remains now headless

PiiXiieeS committed:
fix: move docs images to images folder

MeStrak commented on src/proxy-api/proxy-api.service.ts:
```diff
@@ -57,8 +58,10 @@ export class ProxyApiService {
         const atlassianIadcRegEx = new RegExp(`${baseURL}/wiki/`);
-        parseHeaderBlog(doc.content.body.view.value)(this.context);
+        parseHeaderBlog()(this.context);
+        // getFirstExcerpt()(this.context);
```
I can't see anywhere in the code that `getFirstExcerpt()` is called - is this commented out line intended?

PiiXiieeS replied:
Yes, I did not activate this yet, because I wanted to discuss in a future sprint with you and the team, regarding the full search strategy and how much content we collect to make the API useful. While with more content slower.
So far the partseHeaderBlog is retrieving the excerpt and blog image which was the original intend. My idea was to make it more generic, but to come in a future sprint.

MeStrak commented on src/proxy-page/proxy-page.service.ts:
```diff
@@ -113,12 +117,8 @@ export class ProxyPageService {
-      addNoZoom()(this.context);
-    }
-    addHighlightjs(this.config)(this.context);
+    addLibrariesCSS()(this.context);
```
The implementation looks good, but do you have some ideas for an automated test strategy for this new function? (also for `addLibrariesJS()`). Can any of the existing tests be used as a model?

PiiXiieeS replied:
I had the same thought but not with an obvious outcome.
On one hand we could just check that the libraries are properly added in the HTML head and body structure. That is the obvious and simple unit test.
Going beyond may imply some initialization of the libraries or checking some basic and mandatory styles, but this could be quite oriented and to end up quite complex.

MeStrak commented on src/proxy-page/proxy-page.service.ts:
```diff
@@ -113,12 +117,8 @@ export class ProxyPageService {
     }
     delUnnecessaryCode()(this.context);
     addCustomCss(this.config, style)(this.context);
```
Would it make sense to merge `addCustomCss` with `addLibrariesCss` as the two functions are so similar? Or is the intention to keep `addLibrariesCss` dedicated to external css libraries?

PiiXiieeS replied:
Yes, I make it intentional to split internal and external libraries but something we can discuss and align later on if we find the split is not relevant.

kiru42 reviewed (approved):
That's looks good to me, I also tested it locally.

Commit:
test(addLibraries): add some basic unit tests

Commit:
chore: add TODO comment for getFirstExcerpt

Commit:
test: update test description

MeStrak reviewed (approved):
I added some very simple unit tests for the new addLibraries functions.

Looks good to me now.
